crates/ruff_python_parser/src/lexer/cursor.rs

```diff
@@ -3,16 +3,22 @@ use std::str::Chars;
 
 pub(crate) const EOF_CHAR: char = '\0';
 
+/// A cursor represents a pointer in the source code
```

Comment:
I love it how you add documentation as you go! It's very obvious on what code you've been working or that you touched. It's the code that is well documented. Thanks for doing it.